### PR TITLE
Update DG diagram and storage interface 

### LINKED ISSUE #315: Suggestion: Update Storage Interface overrides
According to https://github.com/nus-cs2103-AY2021S2/forum/issues/224, the intention of the `Storage` interface is to see all the storage component interface methods in one place. Therefore, we should add the missing storage interface methods from its parent interfaces.

### PR DESCRIPTION
**Pull request by cnlinh**
### What this does 
Closes #315 

### How to test
- Check if the diagrams have solid lines with arrows for the association.
- Check if `Storage` has all the correct methods.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -232,11 +232,15 @@ Finer details than what is shown in the section below can be seen under the impl
 
 ![Structure of the Storage Component](images/StorageClassDiagram.png)
 
+**Diagram Notes**:
+* `Storage` implements `CommandHistoryStorage`, `UserPrefsStorage` and `AddressBookStorage`. However, this interface inheritance is omitted for brevity.
```
Comment (benedictkhoomw) [inline docs/DeveloperGuide.md:236]: Hmm, should "implements" be "inherits from"? Technically, interfaces cannot implement anything.

**Review by benedictkhoomw — CHANGES_REQUESTED** [review]
Just one nit to fix.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (cnlinh) [reply docs/DeveloperGuide.md]: Updated!

**Review by benedictkhoomw — APPROVED** [review]
LGTM.

**Pull request merged**